benchmarks/benchmarks/bench_function_base.py

```diff
@@ -96,46 +96,152 @@ def time_select_larger(self):
 
 
 class Sort(Benchmark):
+    """
+    This benchmark tests sorting performance with several
+    different types of arrays that are likely to appear in
+    real-world applications.
+
+    ordered
+        Perfectly ordered array
+    reversed
+        A reversed array
+    uniform
+        A array where all values are the same
+    sorted_block_X
+        An array that is entirely composed of sorted blocks
+        of size X
+    swapped_pair_X_percent
+        An array which has random pairs swapped. The number
+        of swapped pairs is X% of the size of the array
+    random_unsorted_area_X_percent
+        This kind of array has random unsorted areas which
+        take up X% of the original array, the size of one
+        area is AREA_SIZE
+    random_bubble_X_fold
+        ?
```

Comment:
@liwt31 I apologize for yet another ping, but I wasn't able to make out what `random_bubble_X_fold` means. Is it a random unsorted area of size `X`?

Reply:
I think I firstly wrote `random_unsorted_area_X_percent` then I found that this way it's hard to eliminate all already-sorted part of the array. Then I wrote `random_bubble_X_fold` which aims to eliminate all already-sorted parts while keeping the array neraly sorted. The (bad) name comes from the idea that these little permutations sized `BUBBLE_SIZE` are like bubbles. The `X` indicates how strong the permutations is. When `X` is larger, the size of the bubble becomes larger.
Sorry for the ambiguous name and confusing code, when I wrote the code I didn't expect someday it'll become part of NumPy.

Reply:
Are you sure the code is correct, in that sense? `a[start:end]` will return a "truncated" version rather than a "wrapped around" version. Should this really be `a[np.arange(start, end) % size]`? All of the time it'll always go up to the end of the array.

Another concern I have is that if shuffling all these integers unsorts them the first time around, how is this method supposed to introduce back partial ordering?

Reply:
IMHO "truncated" or "wrapped around" are implementation details depending on the understanding of "nearly sorted" and both are ok. "wrapped around" is indeed stricter though.
I'm not sure I fully understand your second question. For totally ordered integers there's no need to consider partial ordering, right?